source/common/http/filter/fault_filter.cc

```diff
@@ -78,31 +93,115 @@ FilterHeadersStatus FaultFilter::decodeHeaders(HeaderMap& headers, bool) {
     return FilterHeadersStatus::Continue;
   }
 
-  if (config_->runtime().snapshot().featureEnabled("fault.http.delay.fixed_delay_percent",
-                                                   config_->delayPercent())) {
-    uint64_t duration_ms = config_->runtime().snapshot().getInteger(
-        "fault.http.delay.fixed_duration_ms", config_->delayDuration());
+  if (headers.EnvoyDownstreamServiceCluster()) {
+    downstream_cluster_ = headers.EnvoyDownstreamServiceCluster()->value().c_str();
+
+    downstream_cluster_delay_percent_key_ =
+        fmt::format("fault.http.{}.delay.fixed_delay_percent", downstream_cluster_);
+    downstream_cluster_abort_percent_key_ =
+        fmt::format("fault.http.{}.abort.abort_percent", downstream_cluster_);
+    downstream_cluster_delay_duration_key_ =
+        fmt::format("fault.http.{}.delay.fixed_duration_ms", downstream_cluster_);
+    downstream_cluster_abort_http_status_key_ =
+        fmt::format("fault.http.{}.abort.http_status", downstream_cluster_);
+  }
+
+  if (isDelayEnabled()) {
+    uint64_t duration_ms = delayDuration();
 
     // Delay only if the duration is >0ms
     if (0 != duration_ms) {
```

Comment:
If you are going to provide a isDelayEnabled() utility function, I suggest folding this condition into that utility. Saves an unnecessary if check and makes code simple.